Remove vendor

zmb3 opened the pull request:
This implements #9545. Do not merge until that is approved and merged.

zmb3 committed:
Remove vendor

- Remove the vendor directory
- Update bot to stop accounting for vendor
- Update linter config
- Remove update-vendor make target

greedy52 commented on Makefile:
```diff
@@ -926,31 +924,6 @@ init-submodules-e: init-webapps-submodules-e
-	# update modules in api/
-	cd api && go mod tidy
-	# update modules in root directory
-	go mod tidy
```
i think running [`go mod tidy`](https://go.dev/ref/mod#go-mod-tidy) may still be beneficial to update/fix go.mod and go.sum

zmb3 replied:
Agree - but we don't need a make target for that. Just like we don't have a make target for `go get` when we want to add or update a dependency. Just use the standard `go` commands directly for operations on go.mod.

greedy52 replied:
Yes. i think i was thinking more about the cases where go.mod/sum messed up (like maybe someone manually did something to these files or bad merges). Good thing now go build will fail in those cases. So maybe it's better this way so that CI build fails and that forces ppl to correct them.

zmb3 committed:
Mark RFD 47 as implemented

Joerger committed:
Disable make target update-api-module-path.

zmb3 committed:
Move GOMODCACHE out of workspace

When GOMODCACHE is in the workspace, some of our make targets end
up running tests against non-Teleport Go modules.
